core-bundle/src/Resources/contao/drivers/DC_Table.php

```diff
@@ -4827,6 +4827,23 @@ protected function listView()
 				{
 					$orderBy[$k] = $key . ' ' . $direction;
 				}
+
+				// If the label field contains a reference to another table, join the table for correct sorting
+				foreach ($GLOBALS['TL_DCA'][$this->strTable]['list']['label']['fields'] as $field)
+				{
+					$arrField = explode(':', $field, 2);
+
+					if ($arrField[0] === $key && \count($arrField) === 2)
+					{
+						$fieldRef = explode('.', $arrField[1], 2);
+						$orderBy[$k] = sprintf('(SELECT %s FROM %s WHERE %s.id = %s.%s)', $fieldRef[1], $fieldRef[0], $fieldRef[0], $this->strTable, $key);
```

Comment:
```suggestion
						$orderBy[$k] = sprintf('(SELECT %s FROM %s WHERE %s.id = %s.%s)', Database::quoteIdentifier($fieldRef[1]), $fieldRef[0], $fieldRef[0], $this->strTable, $key);
```

As we do this in the DataContainer that uses these fields too: https://github.com/contao/contao/blob/b1455051078bee153bcfdd7af1b26651c42e6e78/core-bundle/src/Resources/contao/classes/DataContainer.php#L1679-L1683

We might also want change the coding style to be similar to the one from the DataContainer.